File: MATLAB/denavitHartenberg.m
function [DH] = denavitHartenberg(q, L)
% This function computes the Denavit - Hartenberg matrix (see instructinons inside)
%{
    This function computes the Denavit - Hartenberg matrix based on the 
    generalized coordinates «q» and the length of the rigid bodies «L» 
%}
    DH = [0 0 0 0
          q(1) L(1) 0 pi/2
          q(2) 0 L(2) 0
          q(3) 0 0 pi/2
          q(4) L(3) 0 0];
end

File: Python/DenavitHartenberg.py
import numpy as np

def matrix(robot):
    """
      Denavit - Hartenberg parameters for n - th rigid body
      theta: rotation on «z» axis
      d: translation on «z» axis
      a: translation on «x» axis
      alpha: rotation on «x» axis
    """
    return np.array([[0, 0, 0, 0],
                     [robot.jointsPositions[0, 0], robot.linksLengths[0], 0, np.pi / 2],
                     [robot.jointsPositions[1, 0], 0, robot.linksLengths[1], 0],
                     [robot.jointsPositions[2, 0], 0, 0, np.pi / 2],
                     [robot.jointsPositions[3, 0], robot.linksLengths[2], 0, 0]])